### ISSUE TITLE
Crashing Server after a day

### ISSUE THREAD

**Issue by MineTech13** [mailcow/mailcow-dockerized#309]
Hello my server Crashes nerly every day when mailcow is running but when it is stoped my server dont crash pls help

**Comment by chriscroome** [comment]
My first guess would be a lack of RAM, check your logs to see if `oom-killer` is taking services out:

```
grep oom-killer /var/log/*
```

**Comment by MineTech13** [comment]
Server Specs:
CPU: 4x vCores 7199 MHz
RAM: 16GB
Hard Drive: 125GB HDD

**Comment by andristeiner** [comment]
On Debian? Which kernel version? Might be related to this one: https://github.com/moby/moby/issues/29397
I solved this by using kernel version 4.9 from Debian backports.

**Comment by stevesbrain** [comment]
Otherwise, bad RAM?

**Comment by jm3west** [comment]
Yes, I have got the same problem.
VPS-OS Debian 8 Jessie.

Overnight stop the server

**Comment by stevesbrain** [comment]
@jm3west - Possible to use your own kernel with that VPS provider I presume?

**Comment by jm3west** [comment]
My linux version/kernel is 3.16.0-4-amd64 (debian-kernel@lists.debian.org) (gcc version 4.8.4 (Debian 4.8.4-1) ) #1 SMP Debian 3.16.43-2 (2017-04-30)

**Comment by stevesbrain** [comment]
@jm3west - yep, same kernel as referenced in the moby issue above. Does kernel 4.9 solve your issue?

**Comment by MineTech13** [comment]
okay iam not alone with the problem

**Comment by andryyy** [comment]
The solution is to upgrade the kernel. As mentioned in the docs. :)

**Comment by Ar1sC** [comment]
Im having problem upgrading to Kernel 4.9.....

> Some packages could not be installed. This may mean that you have
> requested an impossible situation or if you are using the unstable
> distribution that some required packages have not yet been created
> or been moved out of Incoming.
> The following information may help to resolve the situation:
> 
> The following packages have unmet dependencies:
>  linux-image-4.9.0-0.bpo.3-amd64 : Depends: linux-base (>= 4.3~) but 3.5 is to be installed
> E: Unable to correct problems, you have held broken packages.
> 
My Kernel Version 3.16.0-4-amd64

**Comment by Braintelligence** [comment]
Is the obvious solution to first upgrade to kernel 4.3 or am I misinterpreting things here?

**Comment by Ar1sC** [comment]
@Braintelligence Switched to Ubuntu 16.04. I couldn't find kernel 4.3.

**Comment by Braintelligence** [comment]
Strange, did you try googling your problem?

**Comment by Ar1sC** [comment]
I Did

**Comment by Braintelligence** [comment]
What happens if you follow this guide?

Please post errors.

http://www.ubuntumaniac.com/2015/12/install-linux-kernel-433-on-ubuntu-1604.html

**Comment by Ar1sC** [comment]
Thank's for your help but Im Already Working on Ubuntu....

**Comment by Braintelligence** [comment]
I don't understand your answer.
Using Ubuntu doesn't make you magically use a specific kernel.
You can update your kernel to a version of your liking as long as you follow the upgrade path and rely on official repositories.

**Comment by stevesbrain** [comment]
@Ar1sC Have you tried installing from these:

http://kernel.ubuntu.com/~kernel-ppa/mainline/v4.9/

If so, have you tried building from source instead?

**Comment by jm3west** [comment]
For Debian .....

echo "deb http://ftp.debian.org/debian jessie-backports main" >> /etc/apt/sources.list
apt-get update
apt-get install -t jessie-backports linux-{image,headers}-amd64


.... and have fun

**Comment by catcun** [comment]
I have the same Debian/Docker problem. I did upgrade the kernel but docker complained and I was not able to make it work; in the end I rolled back the VM. I will migrate the server to Ubuntu this weekend.

**Comment by chriscroome** [comment]
I haven't have any problems running Docker / Mailcow on [Debian Stretch](https://wiki.debian.org/DebianStretch) with linux-image-4.9.0-3-amd64 and with Docker installed following [these instructions](https://docs.docker.com/engine/installation/linux/debian/#install-using-the-repository).

**Comment by Ar1sC** [comment]
@jm3west Yep. It's Working like a charm.
Tested on Debian 8.8.

**Comment by jm3west** [comment]
@Ar1sC, your welcome. My friend :-)

**Comment by Ar1sC** [comment]
@jm3west 
echo "deb http://ftp.debian.org/debian jessie-backports main" >> /etc/apt/sources-list
Edit it to
echo "deb http://ftp.debian.org/debian jessie-backports main" >> /etc/apt/sources.list

you had a typo 👍

**Comment by jm3west** [comment]
ups, I'm sorry.

I've got change the mistake

**Comment by elvirdz** [comment]
@catcun 
after kernel update on 4.9 you have to change docker on overlay2, aufs not work with 4.9 kernel
stop the docker and remove aufs from docker folder

**Comment by catcun** [comment]
@elvirdz
Thanks for the suggestion. I still have a VM on debian, I may try

**Comment by spacie2136** [comment]
Is anybody trying to fix this issue? I need and want to use Kernel 5.3. :-(

**Comment by andryyy** [comment]
This is not an issue anymore. Are you using Jessie with 5.3?

You don't need aufs. Use overlayfs.

**Comment by spacie2136** [comment]
I am using buster/sid with 5.3.0-1030-gcp. I got the problem that the server freezes after some days since i installed Mailcow.

**Comment by andryyy** [comment]
Okay, but there is really nothing I can do with this information. :(
